eth_worker/eth_src/sql_persistence/interface.py

```diff
@@ -508,6 +512,76 @@ def check_if_synchronization_filter_exists(self, contract_address, filter_parame
             return False
         return filter
 
+    def get_transaction_sync_metrics(self):
+        def __query_tuple_list_to_dict__(result):
+            return { key: value for (key, value) in result }
+
+        unsynchronized_transaction_count = self.session.query(
+            BlockchainTransaction.contract_address, func.count(BlockchainTransaction.id))\
+            .group_by(BlockchainTransaction.contract_address)\
+            .filter(BlockchainTransaction.is_synchronized_with_app == False)\
+            .all()
+
+        synchronized_transaction_count = self.session.query(
+            BlockchainTransaction.contract_address, func.count(BlockchainTransaction.id))\
+            .group_by(BlockchainTransaction.contract_address)\
+            .filter(BlockchainTransaction.is_synchronized_with_app == True)\
+            .all()
+
+        unsynchronized_block_count = self.session.query(
+            SynchronizationFilter.contract_address, func.count(SynchronizedBlock.id))\
+            .group_by(SynchronizationFilter.contract_address)\
+            .filter(SynchronizedBlock.status != 'SUCCESS')\
+            .join(SynchronizationFilter)\
+            .all()
+
+        synchronized_block_count = self.session.query(
+            SynchronizationFilter.contract_address, func.count(SynchronizedBlock.id))\
+            .group_by(SynchronizationFilter.contract_address)\
+            .filter(SynchronizedBlock.status == 'SUCCESS')\
+            .join(SynchronizationFilter)\
+            .all()
+
+        max_synchronized_blocks = self.session.query(SynchronizationFilter.contract_address, SynchronizationFilter.max_block).all()
+        
+        last_time_synchronized = self.session.query(SynchronizationFilter.contract_address, SynchronizationFilter.updated).all()
+
+        self.get_failed_callbacks()
```

Comment:
what does this do? the return doesn't seem to be captured

Reply:
Good catch!